--- bochs-2.2.6/gdbstub.cc	2006-01-17 12:15:29.000000000 -0500
+++ bochs-2.2.6-patched/gdbstub.cc	2007-02-06 23:04:51.095523904 -0500
@@ -515,7 +515,7 @@
                         }
                       else
                         {
-                           put_reply("ENN");
+                           put_reply("Eff");
                         }
                    }
                  break;                    
@@ -761,7 +761,7 @@
               }
             else
               {
-                 put_reply("ENN");
+                 put_reply("Eff");
               }
             break;
             
@@ -782,7 +782,7 @@
               }
             else
               {
-                 put_reply("ENN");
+                 put_reply("Eff");
               }          
             break;
 
